lld/MachO/ObjC.cpp

```diff
@@ -1188,6 +1200,36 @@ void ObjcCategoryMerger::eraseMergedCategories() {
                                   catLayout.instancePropsOffset);
     }
   }
+
+  removeRefsToErasedIsecs();
+}
+
+// The compiler may generate references to categories inside the addrsig
+// section. This function will erase these references.
+void ObjcCategoryMerger::removeRefsToErasedIsecs() {
+  for (InputSection *isec : inputSections) {
+    if (isec->getName() != section_names::addrSig)
+      continue;
+
+    auto &_erasedIsecs = erasedIsecs;
```

Comment:
I'm not sure I understand why we need a copy here. Does this not work if we use `erasedIsecs` directly?

Reply:
Can't capture `erasedIsecs` because `'erasedIsecs' in capture list does not name a variable` , but I can fully capture `this` to not use temp variable.